flow/surface_frame.h

```diff
@@ -106,8 +114,17 @@ class SurfaceFrame {
 
   sk_sp<DisplayList> BuildDisplayList();
 
+  void set_user_data(std::shared_ptr<impeller::Surface> data) {
```

Comment:
This is a little wonky. For the impeller metal gpu surface, we construct the corresponding render target during the encode callback and not before - due to partial repaint - so we need a place to store the shared ptr for the present callback.